RCTAppleHealthKit/RCTAppleHealthKit+Queries.m

```diff
@@ -651,4 +652,51 @@ - (void)fetchCumulativeSumStatisticsCollection:(HKQuantityType *)quantityType
     [self.healthStore executeQuery:query];
 }
 
+ - (void)fetchWorkoutForPredicate: (NSPredicate *)predicate
+                       ascending: (BOOL)ascending
+                           limit:(NSUInteger)limit
+                      completion:(void (^)(NSArray *, NSError *))completion {
+
+    void (^handlerBlock)(HKSampleQuery *query, NSArray *results, NSError *error);
+    NSSortDescriptor *endDateSortDescriptor = [[NSSortDescriptor alloc] initWithKey:HKSampleSortIdentifierEndDate ascending:ascending];
+    handlerBlock = ^(HKSampleQuery *query, NSArray *results, NSError *error) {
+        if(!results) {
+            if(completion) {
+                completion(nil, error);
+            }
+            return;
+        }
+
+        if(completion) {
+            NSMutableArray *data = [NSMutableArray arrayWithCapacity:1];
+            NSDictionary *numberToWorkoutNameDictionary = [RCTAppleHealthKit getNumberToWorkoutNameDictionary];
+
+            dispatch_async(dispatch_get_main_queue(), ^{
+                for (HKWorkout * sample in results) {
+                    double energy = [[sample totalEnergyBurned] doubleValueForUnit:[HKUnit kilocalorieUnit]];
+                    double distance = [[sample totalDistance] doubleValueForUnit:[HKUnit mileUnit]];
+                    NSNumber *activityNumber =  [NSNumber numberWithInt: [sample workoutActivityType]];
+
+                    NSDictionary *elem = @{
+                                           @"activityName" : [numberToWorkoutNameDictionary objectForKey: activityNumber],
+                                           @"calories" : @(energy),
+                                           @"distance" : @(distance),
+                                           @"startDate" : [RCTAppleHealthKit buildISO8601StringFromDate:sample.startDate],
```

Comment:
@skryl @bogdanvidican https://app.bugsnag.com/nutrisense/nutrisense-ios/errors/5e59ac5ef343bc00194abad3?filters[event.since][0]=30d&filters[error.status][0]=open&filters[user.id][0]=23&pivot_tab=event